consensus/bor/snapshot.go

```diff
@@ -200,6 +186,40 @@ func (s *Snapshot) apply(headers []*types.Header) (*Snapshot, error) {
 	return snap, nil
 }
 
+// getSignerSuccessionNumber returns the relative position of signer in terms of the in-turn proposer
+func (s *Snapshot) getSignerSuccessionNumber(signer common.Address, proposer common.Address) (int, error) {
```

Comment:
We should write test cases for this function as it is error-prone.

Reply:
Done.